pact/provider/core/src/main/java/org/arquillian/algeron/pact/provider/core/ArgumentPatternMatcher.java

```diff
@@ -0,0 +1,32 @@
+package org.arquillian.algeron.pact.provider.core;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ArgumentPatternMatcher {
+
+    public ArgumentPatternMatcher() {
+    }
+
+    public static List<String> arguments(Pattern pattern, String state) {
+        final List<String> argumentsValues = new ArrayList<>();
+        Matcher matcher = pattern.matcher(state);
+        if (matcher.lookingAt()) {
+            for (int i = 1; i <= matcher.groupCount(); i++) {
+                int startIndex = matcher.start(i);
+
+                if (startIndex == -1) {
+                    argumentsValues.add(null);
```

Comment:
it is handled this null value as well, but let's do an empty string to be more safe to NPE